src/coral_models/prepare_raw_data.py

```diff
@@ -138,32 +151,32 @@ def make_recording_metadata(
                 ]
             )
         ]
-        + ["subject_a_mail", "subject_b_mail", "recorder_mail"]
+        + cols_needed_for_ids
     ]
 
-    # Make a dict from speaker mail to speaker id in the speakers dataframe
-    speaker_mail_to_id = dict(zip(speakers["mail"], speakers["speaker_id"]))
-
     # Make a speaker_id column for the recording metadata, with ids separated by a comma
     # Subject A is the first speaker id, subject B is the second speaker id
     recording_metadata["speaker_id"] = recording_metadata[
-        ["subject_a_mail", "subject_b_mail"]
-    ].apply(lambda x: ",".join([str(speaker_mail_to_id[mail]) for mail in x]), axis=1)
+        ["subject_a_name", "subject_a_mail", "subject_b_name", "subject_b_mail"]
+    ].apply(lambda x: (speaker_id(x[0], x[1]) + "," + speaker_id(x[2], x[3])), axis=1)
```

Comment:
Is this for conversational data, since there are two subjects? If so, can we make this more clear?